x-pack/legacy/plugins/ml/server/new_platform/data_analytics_schema.ts

```diff
@@ -13,7 +13,8 @@ export const dataAnalyticsJobConfigSchema = {
     results_field: schema.maybe(schema.string()),
   }),
   source: schema.object({
-    index: schema.string(),
+    index: schema.oneOf([schema.string(), schema.arrayOf(schema.string())]),
+    query: schema.maybe(schema.any()),
```

Comment:
I think we should add also add in `_source` here, as otherwise adding in e.g.

```
"_source": { 
      "includes": [],
      "excludes": [
        "FlightDelay",
        "FlightDelayType"
      ]
    }
```
via the advanced editor will also cause an error.

Either `_source: schema.maybe(schema.any()),` or something like

```
_source: schema.maybe(
    schema.object({
      includes: schema.maybe(schema.arrayOf(schema.string())),
      excludes: schema.maybe(schema.arrayOf(schema.string())),
    })
  ),
```

And looking at the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.6/put-dfanalytics.html) , we should probably also add the optional `allow_lazy_start` prop.

This highlights that adding in schema validation means that we will need to be very careful in future to watch out for new properties added on the back-end to the request body, which would cause errors in the advanced editor.

Reply:
Good catch! Added the extra definitions to the schema here: ec225be1d71812b511e97a0d92f8142fff7e4d8e

Agreed - I think we should chat about making sure we get pinged for review when the request body changes on the back end. 🤔